autoload/neomake/signs.vim

```diff
@@ -46,25 +56,19 @@ function! neomake#signs#PlaceSign(entry) abort
     endif
 endfunction
 
-function! neomake#signs#CleanOldSigns() abort
-    if !empty(s:last_placed_signs)
-        call neomake#utils#DebugObject('Cleaning old signs:', s:last_placed_signs)
+function! neomake#signs#CleanOldSigns(bufnr) abort
+    if !has_key(s:last_placed_signs, a:bufnr)
```

Comment:
Could change to:

```
if !has_key(s:last_placed_signs, a:bufnr) || empty(s:last_placed_signs[a:bufnr])
    return
endif
```

Then you wouldn't have to wrap the DebugObject call in the if statement.